internal/exec/engine.go

```diff
@@ -103,6 +110,41 @@ func (e Engine) Run(stageName string) error {
 	return nil
 }
 
+// isConfigCacheExist checks if we already have a cached config.
+// TODO(sohan): This is a hack to emit the journald message only
+// once for the base config (https://github.com/coreos/ignition/issues/950)
+func (e *Engine) isConfigCacheExist(cache string) error {
+	var haveConfig bool
+	if _, err := os.Stat(e.ConfigCache); err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		haveConfig = true
+	}
+	if !haveConfig {
+		if err := logStructuredJournalEntry("base", "system", "false"); err != nil {
+			return err
+		}
+	}
```

Comment:
Hmm, I don't follow why this is here. Shouldn't that be in `Run` and only do the logging if we didn't have a cached config (as a sign that it's a first run) but we *did* have a base config?

Reply:
Yeah, this code is logging the base config details if we didn't encounter a cached config. But I would agree with you that for simplicity, I will place this logic in `Run`